ui/withApollo.tsx

```diff
@@ -0,0 +1,37 @@
+import { ApolloClient, InMemoryCache, HttpLink, NormalizedCacheObject } from 'apollo-boost/lib/index'
+
+let apolloClient: ApolloClient<NormalizedCacheObject> | null = null
+
+export const create = (initialState?: NormalizedCacheObject): ApolloClient<NormalizedCacheObject> => {
+  return new ApolloClient({
+    connectToDevTools: process.browser,
+    ssrMode: !process.browser, // Disables forceFetch on the server (so queries are only run once)
+    link: new HttpLink({
+      uri: 'https://us-central1-cloud-computing-5859.cloudfunctions.net/api',
```

Comment:
`/api` should be fine. You def don't want to hard-code the domain here.

Reply:
With Server side rendering and Apollo Client you need to have the full URL in Httplink

Reply:
Do firebase functions inject an env var with the current domain? Def can't hard-code it here. Worst-case, there is a config or .env file somewhere with this so it can be easily over-ridden. Better if it can be generic though.

Reply:
Moved APIURL to `ui/config` I attempted to use Firebase functions config but I was unable to get it working.